Form/ChoiceList/PhpcrOdmQueryBuilderLoader.php

```diff
@@ -93,8 +94,13 @@ public function getEntitiesByIds($identifier, array $values)
         $qb = clone $this->queryBuilder;
         $alias = $qb->getPrimaryAlias();
         $where = $qb->andWhere()->orX();
+
         foreach ($values as $val) {
-            $where->same($val, $alias);
+            if (UUIDHelper::isUUID($val)) {
+                $where->eq()->field($alias.'.uuid')->literal($val)->end();
```

Comment:
Here I force it to be on the `uuid` property of the Document. Since the `class` passed in the constructor is sometimes `null`, I cant use its metadata. Any ideas on how we could get the propert metadata here?